packages/story-editor/src/app/media/utils/getResourceFromUrl.js

```diff
@@ -44,37 +45,67 @@ import {
  * @return {Promise<Resource>} Resource object.
  */
 async function getResourceFromUrl(resourceLike) {
-  const { src, mimeType, ...rest } = resourceLike;
+  const {
+    src,
+    mimeType,
+    width,
+    height,
+    isMuted,
+    length,
+    lengthFormatted,
+    ...rest
+  } = resourceLike;
   const type = getTypeFromMime(mimeType);
 
   if (!['image', 'video'].includes(type)) {
     throw new Error('Invalid media type.');
   }
 
-  const isVideo = type === 'video';
+  const hasDimensions = width && height;
+  const videoHasMissingMetadata =
+    !hasDimensions ||
+    isMuted === null ||
+    length === null ||
+    lengthFormatted === null;
+
+  const additionalData = {};
 
-  const getMediaDimensions = isVideo ? getVideoDimensions : getImageDimensions;
-  const { width, height } = await getMediaDimensions(src);
+  // Only need to fetch metadata if not already provided.
+
+  if (type === 'video' && videoHasMissingMetadata) {
+    const video = await preloadVideo(src);
+    await seekVideo(video);
+
+    additionalData.width = video.videoWidth;
+    additionalData.height = video.videoHeight;
+
+    const videoLength = getVideoLength(video);
+
+    additionalData.length = videoLength.length;
+    additionalData.lengthFormatted = videoLength.lengthFormatted;
```

Comment:
```suggestion
    const { length, lengthFormatted } = getVideoLength(video);

    additionalData = {
       ...additionalData,
       length,
       lengthFormatted
    };
```

Reply:
That does not work because `length` and `lengthFormatted` are already declared.